### PR TITLE
implementation version numbers in `bowtie summary` (only if multiple versions of the same implementation were present)

### LINKED ISSUE #1267: `bowtie summary` should output the summary along with version numbers of implementations.
We don't want to do it always, just if the same implementation appears multiple times with different versions like shown below...

```
⊙  bowtie suite -i ruby-json_schemer -i image:ruby-json_schemer:0.2.25 -i image:ruby-json_schemer:0.2.17 7 | bowtie summary --show failures
Ran 252 test cases.
                      Bowtie                       
┏━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┓
┃ Implementation      ┃ Skips ┃ Errors ┃ Failures ┃
┡━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━━╇━━━━━━━━━━┩
│ json_schemer (ruby) │ 0     │ 0      │ 0        │
│ json_schemer (ruby) │ 0     │ 23     │ 2        │
│ json_schemer (ruby) │ 0     │ 24     │ 3        │
└─────────────────────┴───────┴────────┴──────────┘
```

Ref: [#1250 (comment)](https://github.com/bowtie-json-schema/bowtie/pull/1250#issuecomment-2154358694)

### PR DESCRIPTION
**Pull request by adwait-godbole**
Closes #1267 

Fixed in both pretty and markdown outputs (for both `--show validation` and `--show failures`) like this way.... 

```
(env) PS E:\adwait-godbole\bowtie> bowtie suite -i ruby-json_schemer -i image:ruby-json_schemer:v0.2.25 -i image:ruby-json_schemer:v0.2.17 -i python-jsonschema 7 | bowtie summary --show failures

Ran 252 test cases.
                          Bowtie
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┓
┃ Implementation             ┃ Skips ┃ Errors ┃ Failures ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━━╇━━━━━━━━━━┩
│ json_schemer 2.1.1 (ruby)  │ 0     │ 0      │ 0        │
│ jsonschema (python)        │ 0     │ 0      │ 0        │
│ json_schemer 0.2.25 (ruby) │ 0     │ 23     │ 2        │
│ json_schemer 0.2.17 (ruby) │ 0     │ 24     │ 3        │
└────────────────────────────┴───────┴────────┴──────────┘
                      906 tests ran

(env) PS E:\adwait-godbole\bowtie> bowtie suite -i ruby-json_schemer -i image:ruby-json_schemer:v0.2.25 -i image:ruby-json_schemer:v0.2.17 -i python-jsonschema 7 | bowtie summary --show failures --format markdown

Ran 252 test cases.
# Bowtie Failures Summary

| Implementation | Skips | Errors | Failures |
|:--------------------------:|:-:|:--:|:-:|
| json_schemer 2.1.1 (ruby)  | 0 | 0  | 0 |
|    jsonschema (python)     | 0 | 0  | 0 |
| json_schemer 0.2.25 (ruby) | 0 | 23 | 2 |
| json_schemer 0.2.17 (ruby) | 0 | 24 | 3 |

**906 tests ran**
```
-----------------------------------------------------------------------------------------------------------------------------------------

However, @Julian in the below JSON formatted output, although we are not showing the version number explicitly as in the above outputs but as we can see that in this JSON output, the version number is part of the `implementation.id` (e.g. "image:ruby-json_schemer:v0.2.25") but I am not sure if that version number in that **id** would be always be present since that image tag/id is completely dependent on what the user names it. So it may happen in some cases that the user might likely forget/opt out from stating some version number in that image tag. So should we also explicitly put in the version number along with the result counts in the below output ?

```
(env) PS E:\adwait-godbole\bowtie> bowtie suite -i ruby-json_schemer -i image:ruby-json_schemer:v0.2.25 -i image:ruby-json_schemer:v0.2.17 7 | bowtie summary --show failures --format json

Ran 252 test cases.
[
  [
    "ruby-json_schemer",
    {
      "failed": 0,
      "errored": 0,
      "skipped": 0
    }
  ],
  [
    "image:ruby-json_schemer:v0.2.25",
    {
      "failed": 2,
      "errored": 23,
      "skipped": 0
    }
  ],
  [
    "image:ruby-json_schemer:v0.2.17",
    {
      "failed": 3,
      "errored": 24,
      "skipped": 0
    }
  ]
]
```

### CODE REVIEW

File: bowtie/_cli.py [hunk 1]
```diff
@@ -649,6 +649,8 @@ def _failure_table(
     report: _report.Report,
     results: list[tuple[ConnectableId, ImplementationInfo, Unsuccessful]],
 ):
+    from collections import Counter
```
Comment (Julian) [inline bowtie/_cli.py:652]: Don't think I see a reason to import this locally. Just move it to the top?

**Comment by Julian** [comment]
Can you add a test? They should be reasonably easy to write now with a direct connectable, have a look at the `miniatures` module, and add two there which claim to be the same implementation but send 2 different version numbers.

**Comment by adwait-godbole** [comment]
> However, @Julian in the below JSON formatted output, although we are not showing the version number explicitly as in the above outputs but as we can see that in this JSON output, the version number is part of the implementation.id (e.g. "image:ruby-json_schemer:v0.2.25") but I am not sure if that version number in that id would be always be present since that image tag/id is completely dependent on what the user names it. So it may happen in some cases that the user might likely forget/opt out from stating some version number in that image tag. So should we also explicitly put in the version number along with the result counts in the below output ?

@Julian any comment on the above ?

**Comment by Julian** [comment]
The `id` should always be unique, it's what the user provided, so in the JSON output we should just emit the id as-is I think -- if I follow what you're asking, yes, sometimes that will be the version, sometimes not, but that's fine -- for images we build and push we'll push tags with the version number.

**Pull request closed without merging**